sql/core/src/test/scala/org/apache/spark/sql/connector/WriteDistributionAndOrderingSuite.scala

```diff
@@ -63,36 +65,51 @@ class WriteDistributionAndOrderingSuite
   private val resolver = conf.resolver
 
   test("ordered distribution and sort with same exprs: append") {
-    checkOrderedDistributionAndSortWithSameExprs("append")
+    Seq(false, true).foreach { skew =>
```

Comment:
If every test needs to run with both skew and non-skew, I'd suggest we keep the old test suite unchanged, and add a new `WriteDistributionAndOrderingWithSkewSuite` which extends the old suite.